win32ss/user/ntuser/keyboard.c

```diff
@@ -1020,10 +1020,12 @@ UserSendKeyboardInput(KEYBDINPUT *pKbdInput, BOOL bInjected)
         }
         else
         {
-            wVk = pKbdInput->wVk & 0xFF;
+            wVk = pKbdInput->wVk;
         }
     }
 
+    wVk &= 0xFF;
```

Comment:
This would AND `VK_PACKET` with 0xFF, in case the `if (pKbdInput->dwFlags & KEYEVENTF_UNICODE)` is taken. Is it what you want?

Reply:
That is an unintended effect, and I was in fact about to propose that I move this line to something like this:
```
if (pKbdInput->dwFlags & KEYEVENTF_UNICODE)
    {
        /* Generate WM_KEYDOWN msg with wParam == VK_PACKET and
           high order word of lParam == pKbdInput->wScan */
        wVk = VK_PACKET;
    }
    else
    {
        wScanCode &= 0x7F;
        if (pKbdInput->dwFlags & KEYEVENTF_SCANCODE)
        {
            /* Don't ignore invalid scan codes */
            wVk = IntVscToVk(wScanCode | (bExt ? 0xE000 : 0), pKbdTbl);
            if (!wVk) /* use 0xFF if vsc is invalid */
                wVk = 0xFF;
        }
        else
        {
            wVk = pKbdInput->wVk;
        }
        wVk &= 0xFF;
    }
```

Reply:
this would be better indeed.